test/validate/eia860_test.py

```diff
@@ -48,3 +50,79 @@ def test_own_eia860(pudl_out_eia, live_dbs):
         raise ValueError(
             f"{pct_missing}% of generators lack complete ownership data."
         )
+
+
+@pytest.mark.xfail(reason="There are 40 known inconsistent generator IDs.")
+def test_generator_id_consistency(pudl_out_eia, live_dbs):
+    """
+    Check if there are any plants that report inconsistent generator IDs.
+
+    There are some instances in which a plant will report generator IDs
+    differently in different years, such that the IDs differ only in terms of
+    the case of letters, or non-alphanumeric characters. This test identifies
+    them. We haven't fixed them yet.
```

Comment:
I am... very tempted to lower case all generator IDs... although this would make our data much much harder to merge with original data. Probably a bad idea but it is so tempting.

Reply:
Yeah, I don't think we can muck with the generator IDs, or nobody will be able to easily integrate the data across other sources. It's is pretty annoying that they haven't enforced some uniformity on these things. Whitespace, capitalization, leading zeroes, non-alphanumeric characters, they all create inconsistency across years. Not a lot, but still... too much. I got a response from the @EIAGov Twitter account when I mentioned this the other day so maybe there's someone inside EIA who would be sympathetic if they knew about the issue.